r/R/K_functions.R

```diff
@@ -2,195 +2,206 @@
+    coefficients[10] / temp_k * Istr * sqrt(Istr) +
+    coefficients[11] / temp_k * Istr^2 +
     log(1 - 0.001005 * sal))
 
   return(KS)
 }
 
 #' Calculate Ksp
 #'
-#' @param p Parameters for K calculation
-#' @param temp_k Temperature (Kelvin)
+#' @param coefficients Coefficients for K calculation
+#' @param temp_c Temperature (Celcius)
 #' @param sal Salinity
 #' @return Ksp
-fn_Ksp <- function(p, temp_k, sal) {
-  Ksp <- 10^(p[1] +
-    p[2] * temp_k +
-    p[3] / temp_k +
-    p[4] * log10(temp_k) +
-    (p[5] + p[6] * temp_k + p[7] / temp_k) * sqrt(sal) +
-    p[8] * sal +
-    p[9] * sal * sqrt(sal))
+calc_Ksp <- function(coefficients, temp_c, sal) {
+  temp_k <- temp_c+273.15
+  Ksp <- 10^(coefficients[1] +
+    coefficients[2] * temp_k +
+    coefficients[3] / temp_k +
+    coefficients[4] * log10(temp_k) +
+    (coefficients[5] + coefficients[6] * temp_k + coefficients[7] / temp_k) * sqrt(sal) +
+    coefficients[8] * sal +
+    coefficients[9] * sal * sqrt(sal))
 
   return(Ksp)
 }
 
 #' Calculate KP1
 #'
-#' @param p Parameters for K calculation
-#' @param temp_k Temperature (Kelvin)
+#' @param coefficients Coefficients for K calculation
+#' @param temp_c Temperature (Celcius)
 #' @param sal Salinity
 #' @return KP1
-fn_KP1 <- function(p, temp_k, sal) {
-  KP <- exp(p[1] / temp_k +
-    p[2] +
-    p[3] * log(temp_k) +
-    (p[4] / temp_k + p[5]) * sqrt(sal) +
-    (p[6] / temp_k + p[7]) * sal)
+calc_KP1 <- function(coefficients, temp_c, sal) {
+  temp_k <- temp_c+273.15
+  KP <- exp(coefficients[1] / temp_k +
+    coefficients[2] +
+    coefficients[3] * log(temp_k) +
+    (coefficients[4] / temp_k + coefficients[5]) * sqrt(sal) +
+    (coefficients[6] / temp_k + coefficients[7]) * sal)
 
   return(KP)
 }
 
 #' Calculate KP2
 #'
-#' @param p Parameters for K calculation
-#' @param temp_k Temperature (Kelvin)
+#' @param coefficients Coefficients for K calculation
+#' @param temp_c Temperature (Celcius)
 #' @param sal Salinity
 #' @return KP2
-fn_KP2 <- function(p, temp_k, sal) {
-  KP <- exp(p[1] / temp_k +
-    p[2] +
-    p[3] * log(temp_k) +
-    (p[4] / temp_k + p[5]) * sqrt(sal) +
-    (p[6] / temp_k + p[7]) * sal)
+calc_KP2 <- function(coefficients, temp_c, sal) {
+  temp_k <- temp_c+273.15
+  KP <- exp(coefficients[1] / temp_k +
+    coefficients[2] +
+    coefficients[3] * log(temp_k) +
+    (coefficients[4] / temp_k + coefficients[5]) * sqrt(sal) +
+    (coefficients[6] / temp_k + coefficients[7]) * sal)
 
   return(KP)
 }
 
 #' Calculate KP3
 #'
-#' @param p Parameters for K calculation
-#' @param temp_k Temperature (Kelvin)
+#' @param coefficients Coefficients for K calculation
+#' @param temp_c Temperature (Celcius)
 #' @param sal Salinity
 #' @return KP3
-fn_KP3 <- function(p, temp_k, sal) {
-  KP3 <- exp(p[1] / temp_k +
-    p[2] +
-    (p[3] / temp_k + p[4]) * sqrt(sal) +
-    (p[5] / temp_k + p[6]) * sal)
+calc_KP3 <- function(coefficients, temp_c, sal) {
+  temp_k <- temp_c+273.15
```

Comment:
Programmatically not so nice to repeat the temp_c to temp_k conversion all the time. What was the _ratio_ for not passing temp_k into the function?

Reply:
Yes. Exported functions all used temp_c for usability reasons. For internal functions I think it is preferable not to do code copy where it is not necessary.

Reply:
With Python and Matlab all functions are exported so would be sensible to have them all as `temp_c`.
Still not quite clear what the downside of the extra line of code is? Why is it preferable not to have it?

Reply:
It is fine. We can keep it that way :)